File: source/lang/js/rabbit_template.js
var Rabbit = {
  "zg2uni" : zg2uni,
  "uni2zg": uni2zg
}

function uni2zg(output) {
  var rule = {{UNI2ZG}};
  return replace_with_rule(rule,output);
}

function zg2uni(output) {
  var rule = {{ZG2UNI}};
  return replace_with_rule(rule,output);
}

function replace_with_rule(rule,output) {

  var max_loop = rule.length;
  for(i=0; i < max_loop; i++) {
    
    var data = rule[i];
    var from = data["from"];
    var to = data["to"];

    var from_regex = new RegExp(from,"g");
    output = output.replace(from_regex,to);
  }

  return output;

}

File: source/lang/ruby/rabbit_template.rb
require 'rabbit/version'
require 'json'

# Rabbit Module
module Rabbit
  # Rabbit Convert Class   +  def uni2zg(unicode)
  # Create an instance as follow  class Converter
  # Rabbit::Converter.new
  class Converter
    def uni2zg(unicode)
      json_data = '{{UNI2ZG}}'
      rule = JSON.parse(json_data)
      return self.replace_with_rule(rule,unicode)
    end

    def zg2uni(zawgyi)
      json_data = '{{ZG2UNI}}'
      rule = JSON.parse(json_data)
      replace_with_rule(rule,zawgyi)
    end

    def replace_with_rule(rule, output)
      rule.each do |data|
        from = Regexp.new data['from']
        output = output.gsub(from, data['to'])
      end
      output
    end
  end
end